app/src/tx_display.c

```diff
@@ -389,6 +389,7 @@ parser_error_t tx_display_query(uint16_t displayIdx,
     strncpy_s(outKey, get_required_root_item(root_index), outKeyLen);
 
     if (!display_cache.root_item_start_token_valid[root_index]) {
+        parser_tx_obj.query.out_val = NULL;
```

Comment:
Maybe someone has a better fix for the dangling reference of the local `tmp_val` variable in `parser_tx_obj` ?